src/index.js

```diff
@@ -20,10 +20,185 @@
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'boolean' }
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let probabilisticSamplerSchema = {
+    'id': '/probabilistic',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'number' }
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let ratelimitingSamplerSchema = {
+    'id': '/ratelimiting',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'number' }
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let remoteSamplerSchema = {
+    'id': '/remote',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'string' },
+        'managerHostPort': {'type': 'string'}
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let jaegerSchema = {
+    'id': '/jaeger',
+    'type': 'object',
+    'required': ['serviceName'],
+    'properties': {
+        'serviceName': {'type': 'string'},
+        'disable': {'type': 'boolean'},
+        'sampler': {
+            'type': 'object',
+            // should be 'oneOf', but that doesn't work properly with this implementation.
+            'anyOf': [
+                {'$ref': 'const'},
+                {'$ref': 'probabilistic'},
+                {'$ref': 'ratelimiting'},
+                {'$ref': 'remote'}
+            ]
+        },
+        'reporter': {
+            'properties': {
+                'logSpans': {'type': 'boolean'},
+                'agentHost': {'type': 'string'},
+                'agentPort': {'type': 'number'},
+                'flushIntervalMs': {'type': 'number'}
+            },
+            'additionalProperties': false
+        }
+    }
+};
+
+let getSampler = (options) => {
+    let type = options.sampler.type;
+    let param = options.sampler.param;
+
+    let sampler;
+    if (type === 'probabilistic') {
+       sampler = new ProbabilisticSampler(param);
+    }
+
+    if (type === 'ratelimiting') {
+        sampler = new RatelimitingSampler(param);
+    }
+
+    if (type === 'const') {
+        sampler = new ConstSampler(param);
+    }
+
+    if (type === 'remote') {
+        sampler = new RemoteSampler(options.serviceName);
+    }
+
+    return sampler;
+}
+
+let initTracer = (options) => {
+    let v = new Validator();
+    v.addSchema(constSamplerSchema);
+    v.addSchema(probabilisticSamplerSchema);
+    v.addSchema(ratelimitingSamplerSchema);
+    v.addSchema(remoteSamplerSchema);
+    v.addSchema(jaegerSchema);
+    v.validate(options, configSchema, {
+        throwError: true
+    });
+
+    options = options.jaeger;
+
+    let reporter = null;
+    let sampler = null;
+    if (options.disable) {
+        reporter = new NoopReporter();
+        sampler = new ConstSampler(false);
```

Comment:
let's log something that tracing is disabled and all other configuration is ignored.

Also, in other libs disable causes a NoopTracer to be used (from opentracing), not Jaeger tracer with null reporter, because they want to disable the whole tracing (including wire serialization). If they only want to disable reporter, we can have another disabled flag under reporter which will turn it to NullReporter.